tls/config.go

```diff
@@ -81,6 +80,49 @@ func NewServerConfig(logger log.Logger, certFile, keyFile, minVersion, maxVersio
 	return tlsCfg, nil
 }
 
+func tlsCipherSuites() map[string]uint16 {
+	cipherSuites := map[string]uint16{}
+
+	for _, suite := range tls.CipherSuites() {
+		cipherSuites[suite.Name] = suite.ID
+	}
+	for _, suite := range tls.InsecureCipherSuites() {
+		cipherSuites[suite.Name] = suite.ID
+	}
+
+	return cipherSuites
+}
+
+func parseTLSVersion(rawTLSVersion string) (uint16, error) {
+	switch rawTLSVersion {
+	case "VersionTLS10": 
+		return tls.VersionTLS10, nil
+	case "VersionTLS11": 
+		return tls.VersionTLS11, nil
+	case "VersionTLS12": 
+		return tls.VersionTLS12, nil
+	case "VersionTLS13": 
+		return tls.VersionTLS13, nil
+	default:
+		return 0, fmt.Errorf("unknown TLSVersion: %s", rawTLSVersion)
+	}
+}
+
+func mapCipherNamesToIDs(rawTLSCipherSuites []string) ([]uint16, error) {
+	cipherSuites := []uint16{}
+	allCipherSuites := tlsCipherSuites()
+
+	for _, name := range cipherSuiteNames {
+		id, ok := allCipherSuites[name]
+		if !ok {
+			return nil, fmt.Errorf("unknown TLSCipherSuite: %s", rawTLSCipherSuite)
+		}
```

Comment:
Build fails due to this I think?
```suggestion
	for _, name := range rawTLSCipherSuites {
		id, ok := allCipherSuites[name]
		if !ok {
			return nil, fmt.Errorf("unknown TLSCipherSuite: %s", name)
		}
```